_source/_posts/2022-07-21-jhipster-ci-cd.md

```diff
@@ -0,0 +1,933 @@
+
+- Type of application: **Microservice application**
+- Root directory: **../**
+- Which applications? (select all)
+- Set up monitoring? **No**
+- Which applications with clustered databases? select **store**
+- Admin password for JHipster Registry: (generate one)
+- Kubernetes namespace: **demo**
+- Docker repository name: **your-dockerhub-username**
+- Command to push Docker image: `docker push`
+- Enable Istio? **No**
+- Kubernetes service type? **LoadBalancer**
+- Use dynamic storage provisioning? **Yes**
+- Use a specific storage class? (leave empty)
+
+### Configure Okta for OIDC authentication
+
+One more configuration step before building the applications, let's configure Okta for authentication. You must run the commands at the root folder of the project.
+
+{% include setup/cli.md type="jhipster" %}
+
+Add the settings from the generated `.okta.env` to `kubernetes/registry-k8s/application-configmap.yml`:
+
+```yml
+data:
+  application.yml: |-
+    ...
+    spring:
+      security:
+        oauth2:
+          client:
+            provider:
+              oidc:
+                issuer-uri: https://<your-okta-domain>/oauth2/default
+            registration:
+              oidc:
+                client-id: <client-id>
+                client-secret: <client-secret>
+```
+
+Enable the OIDC authentication in the `jhipster-registry` service by adding the `oauth2` profile in the `kubernetes/registry-k8s/jhipster-registry.yml` file:
+
+```yml
+- name: SPRING_PROFILES_ACTIVE
+  value: prod,k8s,oauth2
+```
+
+We are going to build the `gateway` and `store` in a continuous integration workflow with CircleCI.
+
+## Set up CI for JHipster with CircleCI
+
+First, create the CircleCI configuration for the `store` microservice and the `gateway`.
+
+```shell
+cd store
+jhipster ci-cd
+```
+```shell
+cd gateway
+jhipster ci-cd
+```
+
+For both applications, choose the following options:
+- What CI/CD pipeline do you want to generate? **CircleCI**
+- What tasks/integrations do you want to include? (none)
+
+You must tweak the generated configuration at `store/.circleci/config.yml`, as the following changes are required for successful execution:
+- Change the execution environment of the workflow, to a dedicated VM, as that is required by the [TestContainers](https://www.testcontainers.org/supported_docker_environment/continuous_integration/circle_ci) dependency in tests. The chosen environment is [LinuxVM](https://circleci.com/developer/machine/image/ubuntu-2004), a standalone Ubuntu 20.04 virtual machine.
+- As the dedicated VM includes docker, the docker installation step in the configuration must be removed.
+- Add a step for building the docker container image.
+- Add a step for pushing the image to DockerHub.
+
+The final `config.yml` must look like this:
+
+```yml
+version: 2.1
+jobs:
+  build:
+    environment:
+      DOCKERHUB_USERNAME: your-dockerhub-username
+      IMAGE_NAME: $DOCKERHUB_USERNAME/store
```

Comment:
I tried this and it doesn't work. https://app.circleci.com/pipelines/github/mraible/spinnaker-store/9/workflows/355be434-e680-4985-becc-d0303cf55746/jobs/9

It seems you can't reference the previous variable so this value should be `your-dockerhub-username/store`.

Reply:
fixed.